src/Series.cpp

```diff
@@ -24,7 +24,18 @@
 #include "openPMD/Series.hpp"
 
 #include <iostream>
-#include <regex>
+
+#if defined(__GNUC__)
+#   if (__GNUC__ == 4 && __GNUC_MINOR__ < 9)
```

Comment:
I have tested this locally by replacing this test with a single `#   if __GNUC__ < 9`